qiskit/transpiler/basepasses.py

```diff
@@ -95,6 +95,45 @@ def is_analysis_pass(self):
         """
         return isinstance(self, AnalysisPass)
 
+    def __call__(self, circuit, property_set=None):
+        """Runs the pass on circuit.
+
+        Args:
+            circuit (QuantumCircuit): the dag on which the pass is run.
+            property_set (PropertySet or dict or None): input/output property set.
+
+        Returns:
+            QuantumCircuit: If on transformation pass, the resulting QuantumCircuit. If analysis
+                   pass, the input circuit.
+        """
+        from qiskit.converters import circuit_to_dag, dag_to_circuit
+        from qiskit.dagcircuit.dagcircuit import DAGCircuit
+
+        property_set_ = None
+        if isinstance(property_set, dict):
+            property_set_ = PropertySet(property_set)
+
+        if isinstance(property_set_, PropertySet):
+            self.property_set = property_set_
+
+        result = self.run(circuit_to_dag(circuit))
+
+        result_circuit = circuit
+
+        if isinstance(property_set, (dict, PropertySet)):
+            property_set.update(self.property_set)
```

Comment:
Maybe also `property_set.clear()`, in cases `self.run` removes entries from the property_set?

Reply:
I don't think that passes should remove properties. Although, I dont thing that's written anywhere and the object does not prevent that. So yeap... let's do that. See ae08e9b6d